config/v1/json.go

```diff
@@ -0,0 +1,42 @@
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+)
+
+// reUUID4 uses a regexp to avoid taking a dependency on a UUID library for now
+var reUUID4 = regexp.MustCompile(`^[a-fA-F\d]{8}-[a-fA-F\d]{4}-4[a-fA-F\d]{3}-[89abAB][a-fA-F\d]{3}-[a-fA-F\d]{12}$`)
```

Comment:
https://github.com/openshift/cluster-version-operator/blob/ffcd66a9b1e8ba7364f7ed46d0cac577e5b49e82/vendor/github.com/google/uuid/uuid.go#L138-L149 is the actual test we want to make.

Reply:
I'm going to move this into the CVO, David is right that client side validation doesn't work in the long run.  We already need to reflect validation errors back to the ClusterVersion status, so we might as well put that there.